src/backend/distributed/executor/multi_utility.c

```diff
@@ -816,6 +832,131 @@ ProcessAlterObjectSchemaStmt(AlterObjectSchemaStmt *alterObjectSchemaStmt,
 }
 
 
+/* TODO: Write function comments */
+static Node *
+ProcessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand, bool isTopLevel)
+{
+	Oid relationId = InvalidOid;
+	List *taskList = NIL;
+
+	if (vacuumStmt->relation == NULL)
+	{
+		return (Node *) vacuumStmt;
+	}
+
+	relationId = RangeVarGetRelid(vacuumStmt->relation, NoLock, false);
```

Comment:
`VACUUM` is itself processed using `NoLock` when looking up the relation, was just trying to follow closely how it works:

> Since we don't take a lock here, the relation might be gone, or the RangeVar might no longer refer to the OID we look up here.  In the former case, VACUUM will do nothing; in the latter case, it will process the OID we looked up here, rather than the new one. Neither is ideal, but there's little practical alternative, since we're going to commit this transaction and begin a new one between now and then.

What are you worried about here, now that the master/local `VACUUM` runs first?